legend-sdlc-server/src/test/java/org/finos/legend/sdlc/server/gitlab/api/server/AbstractGitLabServerApiTest.java

```diff
@@ -0,0 +1,114 @@
+{
+    // Note that for the gitlab.com -based tests, the test member and owner are of the same GitLab account
+    // thus have the same credentials, yet used to create separate GitLab API instances for merge request related tests.
+    static final String TEST_OWNER_USERNAME = System.getenv("GITLAB_USERNAME");
+    static final String TEST_OWNER_PASSWORD = System.getenv("GITLAB_PASSWORD");
+    static final String TEST_MEMBER_USERNAME = TEST_OWNER_USERNAME;
+    static final String TEST_MEMBER_PASSWORD = TEST_OWNER_PASSWORD;
+    static final String TEST_HOST_SCHEME = "https";
+    static final String TEST_HOST_HOST = "gitlab.com";
+    static final Integer TEST_HOST_PORT = null;
+    static final String TEST_HOST_URL = "https://gitlab.com";
+
+    static final BackgroundTaskProcessor backgroundTaskProcessor = new BackgroundTaskProcessor(1);
+
+    static final Logger LOGGER = LoggerFactory.getLogger(AbstractGitLabServerApiTest.class);
+
+    @BeforeClass
+    public static void suiteSetup()
+    {
+        JerseyGuiceUtils.install((s, serviceLocator) -> null);
+    }
+
+    @AfterClass
+    public static void shutDown()
+    {
+        LOGGER.info("Shutting down backgroundTaskProcessor.");
+        backgroundTaskProcessor.shutdown();
+        LOGGER.info("Shut down backgroundTaskProcessor.");
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext for a project owner.
+     * Please note that for the server test, separate instances of GitLabApis are necessary for owner and member for the same test account.
+     *
+     * @return A test GitLabUserContext for a project owner.
+     * @throws LegendSDLCServerException if cannot authenticate to GitLab via OAuth.
+     */
+    protected static GitLabUserContext prepareGitLabOwnerUserContext() throws LegendSDLCServerException
+    {
+        return prepareGitLabUserContextHelper(TEST_OWNER_USERNAME, TEST_OWNER_PASSWORD);
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext for a project member.
+     *
+     * @return A test GitLabUserContext for a project member.
+     * @throws LegendSDLCServerException if cannot authenticate to GitLab via OAuth.
+     */
+    protected static GitLabUserContext prepareGitLabMemberUserContext() throws LegendSDLCServerException
+    {
+        return prepareGitLabUserContextHelper(TEST_MEMBER_USERNAME, TEST_MEMBER_PASSWORD);
+    }
+
+    /**
+     * Authenticates to GitLab and creates a test GitLabUserContext.
+     *
+     * @param username the name of user for whom we create this context.
+     * @param password the password of user for whom we create this context.
+     */
+    private static GitLabUserContext prepareGitLabUserContextHelper(String username, String password) throws LegendSDLCServerException
+    {
+        return GitLabApiTestSetupUtil.prepareGitLabUserContextHelper(username, password, TEST_HOST_URL, TEST_HOST_SCHEME, TEST_HOST_HOST, TEST_HOST_PORT, LOGGER);
```

Comment:
It's best not to pass this class's logger to another class. See previous comments.

Reply:
Fixed!